applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -962,14 +977,91 @@ static int usb_init_next_hids_init(void)
 	return err;
 }
 
+static bool is_usb_req_pending(void)
+{
+	return k_event_test(&usb_event, USB_REQ_ENABLE | USB_REQ_DISABLE);
+}
+
+static int handle_usbd_state_on_status_change(enum usbd_msg_type type)
+{
+	switch (type) {
+	case USBD_MSG_VBUS_READY:
+		if (!usb_enabled) {
+			uint32_t events;
+
+			__ASSERT_NO_MSG(!is_usb_req_pending());
+
+			LOG_DBG("Enabling USB");
+			k_event_set(&usb_event, USB_REQ_ENABLE);
```

Comment:
Can we make a common function like `usb_thread_req_send(bool usb_enable)` to avoid code duplication here? Then you could even remove `is_usb_req_pending` because it would be called only once (whole function explicitly uses `USB_REQ_...` flags so I would suggest not to use additional wrapper for the assertion too)